client-metrics/src/main/scala/org/http4s/client/metrics/Metrics.scala

```diff
@@ -0,0 +1,71 @@
+package org.http4s.client.metrics
+
+import cats.data.Kleisli
+import cats.effect._
+import cats.implicits._
+import com.codahale.metrics.{Counter, MetricRegistry, Timer}
+import java.util.concurrent.TimeUnit
+import org.http4s.{Request, Status}
+import org.http4s.client.{Client, DisposableResponse}
+
+object Metrics {
+  def apply[F[_]: Sync](registry: MetricRegistry, prefix: String = "org.http4s.client")(
+      client: Client[F]): Client[F] = {
```

Comment:
This is similar to a conversation we're having in #1969: if we added a `Timer[F]` instead of calling `System.nanoTime`, we could parameterize the clock and thus be able to test the response times.

Reply:
Sure, that makes total sense. Talking about tests, I have not seen any tests in the other metrics middleware. Is it desired that they come with no tests?

Reply:
Sure, that makes total sense. Talking about tests, I have not seen any tests in the other metrics middleware. Is it desired that they come with no tests?

Reply:
Timer didn't exist, so those tests would have been harder to write.  And I think we just got lazy on the other parts.  Tests would be great, but not mandatory.